src/server.c

```diff
@@ -2800,6 +2799,13 @@ int isContainerCommandBySds(sds s) {
     return has_subcommands;
 }
 
+struct redisCommand *lookupSubCommandByCStringLogic(dict *commands, const char *parent, const char *sub) {
```

Comment:
i think a better API would be:
```
struct redisCommand *lookupSubcommand(struct redisCommand *container, const char *sub);
```

Reply:
All done, the second commit:
- change function `moduleCreateCommandProxy` take `sds fullname`
- add `catSubCommandFullname` function do the `sdscatfmt(sdsempty(), "%s|%s", parent_name, sub_name)` thing
- some typos, `setImplictACLCategories` -> `setImplicitACLCategories`, `stracture` -> `structure`
- add `lookupSubcommandByFullname` / `lookupSubcommand` functions
- remove `getFullCommandName` dead function and `populateSingleCommand` dead code